src/service_providers/authentication/decode_token.js

```diff
@@ -1,5 +1,13 @@
+const jwt = require('jsonwebtoken')
+
 function decodeToken (token) {
-  throw new Error('Not implemented!')
+  const verify = jwt.verify(token, 'secret')
+  if (verify) {
+    const decoded = jwt.decode(token, { complete: true })
+    return (decoded.payload)
+  } else {
+    console.log('Error')
```

Comment:
Don't log errors. Use `throw new Error('Some descriptive message')`, it'll allow the caller of your function to do something useful if an error occurs.